tilequeue/store.py

```diff
@@ -2,20 +2,54 @@
 
 from boto import connect_s3
 from boto.s3.bucket import Bucket
-from TileStache.S3 import tile_key
+import md5
 import os
 
 
+def calc_hash(s):
+    m = md5.new()
+    m.update(s)
+    md5_hash = m.hexdigest()
+    return md5_hash[:5]
+
+
+def s3_tile_key(date, path, layer, coord, extension):
+    path_to_hash = '/%(path)s/%(layer)s/%(z)d/%(x)d/%(y)d.%(ext)s' % dict(
+        path=path,
+        layer=layer,
+        z=coord.zoom,
+        x=coord.column,
+        y=coord.row,
+        ext=extension,
+    )
+    md5_hash = calc_hash(path_to_hash)
+    s3_path = ('%(date)s/%(md5)s/'
+               '%(path)s/%(layer)s/%(z)d/%(x)d/%(y)d.%(ext)s' % dict(
+                   path=path,
+                   layer=layer,
+                   z=coord.zoom,
+                   x=coord.column,
+                   y=coord.row,
+                   ext=extension,
+                   date=date,
+                   md5=md5_hash,
+               ))
+    return s3_path
+
+
 class S3(object):
 
-    def __init__(self, bucket, layer_name, path='', reduced_redundancy=False):
+    def __init__(
+            self, bucket, layer_name, date_prefix, path, reduced_redundancy):
```

Comment:
Oh, wait, yes. I see. I should have read all the way to the bottom before commenting.